### PR TITLE
TypeScript Declarations

### PR DESCRIPTION
**Pull request by pvoisin**
Here's an attempt to ship TypeScript declarations with the module so it can be used in TypeScript projects.

Also, while trying out existing code and browsing the source files I found little issues that I've fixed.

### CODE REVIEW

**Comment by pvoisin** [comment]
@dengelke - Whenever you have few minutes... :)

**Comment by dengelke** [comment]
Thanks @pvoisin will give it a look!

File: lib/record.js [hunk 1]
```diff
@@ -8,11 +8,12 @@ function Record(record) {
 
 Record.prototype = {
   get: function (type) {
-    return this._record.get(type);
+    return this._record?.get(type);
```
[the remark this answers is not in the document]
Comment (pvoisin) [reply lib/record.js:11]: Ah, snap! Let me change this to something backward-compatible then...

[the remark this answers is not in the document]
Comment (pvoisin) [reply lib/record.js]: I just remade the commit - hopefully it works now!

[the remark this answers is not in the document]
Comment (dengelke) [reply lib/record.js]: Looks good! Thanks for resolving

**Pull request merged**